module/zcommon/zfs_fletcher.c

```diff
@@ -245,7 +214,107 @@ fletcher_4_incremental_byteswap(const void *buf, uint64_t size,
 	ZIO_SET_CHECKSUM(zcp, a, b, c, d);
 }
 
+static const struct fletcher_4_calls fletcher_4_generic_calls = {
+	.init = fletcher_4_generic_init,
+	.compute = fletcher_4_generic,
+	.compute_byteswap = fletcher_4_generic_byteswap,
+	.name = "generic"
+};
+
+static const struct fletcher_4_calls *chosen = &fletcher_4_generic_calls;
```

Comment:
Let's have a more descriptive name for this.  Preferably something consistent with ` zfs_raidz_math_impl`  which is what @ironMann settled on in 3d17e09897a106fe25044d05ca225afbcd8b1109 for the vectorized raidz.  How about `zfs_fletcher4_math_impl`.

Similarly we're going to want to expose this as a module option is a safe way so it  can be changed dynamically on a running system.  Again making this roughly consistent with the raidz changes would be good for consistency.  The proposed options in that patch are as follows, and could be adapted to this patch.

```
New zfs module parameters:
- zfs_raidz_math_impl (str): selects the implementation to use
    "fastest" - use the fastest math available
    "original" - use the original raidz code
    "cycle" - cycle trough all available impl for each new vdev_raidz
    "scalar" - new scalar impl
    "sse" - new SSE impl if available
    "avx2" - new AVX2 impl if available
```

We should also make sure this code is exercised by `ztest` in user space as a stress test.  Ideally we want it to be cycling through all the available options currently with whatever test it may be running.  This will help expose any corner cases which may have been overlooked.